core/src/test/java/kafka/testkit/KafkaClusterTestKitTest.java

```diff
@@ -122,4 +126,25 @@ public void testCreateClusterAndCloseWithMultipleLogDirs(boolean combined) {
             fail("failed to init cluster", e);
         }
     }
+
+    @Test
+    public void testCreateClusterWithSpecificBaseDir() {
+        Path baseDirectory = TestUtils.tempDirectory().toPath();
+        try (KafkaClusterTestKit cluster = new KafkaClusterTestKit.Builder(
+                new TestKitNodes.Builder().
+                        setBaseDirectory(baseDirectory).
+                        setNumBrokerNodes(1).
+                        setCombined(true).
+                        setNumControllerNodes(1).build()).build()) {
+            assertEquals(cluster.nodes().baseDirectory(), baseDirectory.toFile().getAbsolutePath());
+            cluster.nodes().controllerNodes().values().forEach(controller -> {
+                assertTrue(Paths.get(controller.metadataDirectory()).startsWith(baseDirectory));
+            });
+            cluster.nodes().brokerNodes().values().forEach(broker -> {
+                assertTrue(Paths.get(broker.metadataDirectory()).startsWith(baseDirectory));
+            });
+        } catch (Exception e) {
```

Comment:
it is find to throw exception directly so could you please remove the catch?

Reply:
Sure, this was copied from the test above. Would you prefer I also remove it there?

Reply:
:+1: Comments should have been addressed in the latest commit